index.js

```diff
@@ -1,9 +1,32 @@
 'use strict';
+const URL = require('url').Url;
 const urlRegex = require('url-regex');
 const normalizeUrl = require('normalize-url');
+const unescape = require('unescape');
 
 module.exports = (str, opts) => {
-	const urls = str.match(urlRegex());
-	const ret = urls ? urls.map(url => normalizeUrl(url.trim().replace(/\.+$/, ''), opts)) : [];
-	return new Set(ret);
+	opts = opts || {};
+	const ret = new Set();
+
+	function _extractQueryParams(url) {
+		const qs = new URL().parse(url, true).query;
```

Comment:
This is not the correct way to do it. You want `(new URL(url)).searchParams`. What you're doing is using the old Node.js `querystring` module.

Reply:
what you suggest only works in node 9 and we dont want to constraint our users...

Reply:
Ah ok, my mistake, I thought it got backported. This is fine then.

Reply:
perfect